kdbg/debugger.cpp

```diff
@@ -1106,6 +1108,16 @@ void KDebugger::parse(CmdQueueItem* cmd, const char* output)
     case DCdisassemble:
 	handleDisassemble(cmd, output);
 	break;
+#ifdef Q_PROCESSOR_X86
+    case DCsetdisassflavor:
+	{
+	    //TODO Review this: What this switch does and do we need
+	    // to place these commands here? We have connect(..) for this.
+	    auto pg {KSharedConfig::openConfig()->group("Preferences")};
+	    handleSetDisassFlavor(pg.readEntry("DisassemblyFlavor"));
+	}
```

Comment:
The configuration need not be stored here; that should happen after the dialog was closed. Is there no precedent with other options?

Reply:
Yes the config doesn't need to be stored here, but what I have to write inside this case?

Should I call `handleSetDisassFlavor()` which in turn should call `executeCmdString(DCsetdisassflavor, flavorCStr, false)`?

Reply:
No. This case arm processes the result of the command after Gdb has processed it. I guess that there is nothing to do here. You could look out for an error message: if Gdb accepted the command, all is well; if it didn't, nothing bad happens if the old flavor is still in place. But I don't think that needs to be done in the first step.

`queueCmdString(DCsetdisassflavor, ...)` should be called in `setDisassemblyFlavor()`, I think, and as part of the sequence where the program is loaded, see `KDebugger::debugProgram()`.